Packs/TOPdesk/Integrations/TOPdesk/TOPdesk_test.py

```diff
@@ -0,0 +1,816 @@
+     True,
+     {"query": 'caller_id==some_caller', "status": "firstLine"},
+     'https://test.com/api/v1/incidents?query=caller_id==some_caller&status==firstLine'),
+    (get_incidents_list_command,
+     True,
+     {"query": 'status==firstLine', "caller_id": "some_caller_id", "branch_id": "some_branch"},
+     'https://test.com/api/v1/incidents?query=status==firstLine&caller==some_caller_id&branch==some_branch')
+])
+def test_old_new_query(requests_mock,
+                       command,
+                       new_query,
+                       command_args,
+                       command_api_request):
+    """Unit test
+    Given
+        - command args
+        - which type of query is supported
+    When
+        - running the command
+    Then
+        - validate the correct request url was called.
+    """
+    client = Client(
+        base_url='https://test.com/api/v1',
+        verify=False,
+        auth=('some_username', 'some_password'),
+        new_query=new_query
+    )
+    requests_mock.get(command_api_request, json=[{}])
+    command(client=client, args=command_args)
+
+    assert requests_mock.called
+
+
+@pytest.mark.parametrize('command_args', [
+    ({"category": "blah"}), ({"subcategory": "blah"}), ({"call_type": "blah"}), ({"entry_type": "blah"})
+])
+def test_unsupported_old_query_param(command_args):
+    """Unit test
+    Given
+        - get_incidents_list_command old query setting unsupported command args
+    When
+        - running the command with old query setting
+    Then
+        - validate KeyError is raised.
+    """
+    client = Client(
+        base_url='https://test.com/api/v1',
+        verify=False,
+        auth=('some_username', 'some_password'),
+        new_query=False
+    )
+    with pytest.raises(KeyError, match="is not supported with old query setting."):
+        get_incidents_list_command(client=client, args=command_args)
+
+
+@pytest.mark.parametrize('topdesk_incidents_override, last_fetch_time, updated_fetch_time', [
+    ([{  # Last fetch is before incident creation
+        'number': 'TEST-1',
+        'creationDate': '2020-02-10T06:32:36Z',
+        'will_be_fetched': True
+    }], '2020-01-11T06:32:36.303000+0000', '2020-02-10T06:32:36.000000+0000'),
+    ([{  # Last fetch is after one incident creation and before other.
+        'number': 'TEST-1',
+        'creationDate': '2020-01-10T06:32:36Z',
+        'will_be_fetched': False
+    }, {
+        'number': 'TEST-2',
+        'creationDate': '2020-03-10T06:32:36Z',
+        'will_be_fetched': True
+    }], '2020-02-11T06:32:36.303000+0000', '2020-03-10T06:32:36.000000+0000'),
+    ([{  # Last fetch is at incident creation
+        'number': 'TEST-1',
+        'creationDate': '2020-02-10T06:32:36.303+0000',
+        'will_be_fetched': False
+    }], '2020-02-10T06:32:36.303000+0000', '2020-02-10T06:32:36.303000+0000'),
+])
+def test_fetch_incidents(requests_mock, topdesk_incidents_override, last_fetch_time, updated_fetch_time):
```

Comment:
does it make sense to add test for 2 consecutive calls to `fetch_incidents` (could be same data returned) to ensure duplicates are not fetched?